ui/src/app/services/status-check.service.ts

```diff
@@ -0,0 +1,46 @@
+import {Injectable} from '@angular/core';
+import {Http} from '@angular/http';
+import {Observable} from 'rxjs/Observable';
+
+import {InterceptedHttp} from 'app/factory/InterceptedHttp';
+import {ErrorHandlingService} from 'app/services/error-handling.service';
+
+import {StatusService} from 'generated';
+
+@Injectable()
+export class StatusCheckService {
+  public apiDown: boolean;
+  public firecloudDown: boolean;
+  public notebooksDown: boolean;
+  private obs: Observable<boolean>;
+
+  constructor(
+    private http: Http,
+    private statusService: StatusService,
+  ) {
+    this.apiDown = false;
+    this.firecloudDown = false;
+    this.notebooksDown = false;
+    this.obs = (<InterceptedHttp> http).getStatusSubjectAsObservable();
+    this.obs.subscribe(() => {
+      this.getApiStatus();
+    });
+  }
+
+  private getApiStatus(): void {
+    this.statusService.getStatus().subscribe((resp) => {
```

Comment:
Possibly `(<InterceptedHttp> this.http).shouldPingStatus = false;` would be better placed above this line (as discussed in person).